src/pool/srv_util.c

```diff
@@ -256,7 +256,7 @@ ds_pool_map_tgts_update(struct pool_map *map, struct pool_target_id_list *tgts,
 			D_PRINT("Target (rank %u idx %u) is down.\n",
 				target->ta_comp.co_rank,
 				target->ta_comp.co_index);
-			if (pool_map_node_status_match(dom,
+			if (evict_rank && pool_map_node_status_match(dom,
```

Comment:
You added a new parameter 'evict_rank' to inform this function to mark the node as DOWN, then the pool_map_node_status_match() could be removed.